apps/prairielearn/src/lib/server-jobs.ts

```diff
@@ -214,20 +215,26 @@ class ServerJobImpl implements ServerJob, ServerJobExecutor {
     }
   }
 
-  private addToOutput(msg: string) {
+  private addToOutput(msg: string, forceAdd = false) {
```

Comment:
`forceAdd` isn't a great name; really this is more of a "force flush to socket".

How about this: move the socket stuff to a `private flush(force = false) { ... }` function, then call that both here and in `finish()`?

Reply:
I have another idea, not sure if it's better or not: 

We can remove the `force` parameter so it will always flush, and we call `flush` in `addToOutput` when the time difference is large enough, or call `flush` directly in `finish()` without having to specify a `force=true`?

Reply:
I lean towards having `addToOutput` being unaware of throttling/etc., but I don't have super strong opinions here.

Reply:
I've dated using your original idea.